packages/amplify-codegen/tests/commands/models.test.js

```diff
@@ -71,43 +72,148 @@ describe('command - models', () => {
+            mockFs(mockedFiles);
             MOCK_CONTEXT.amplify.getProjectConfig.mockReturnValue({frontend: frontend});
-            const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
+
             // assert empty folder before generation
             expect(fs.readdirSync(outputDirectory).length).toEqual(0);
+
             await generateModels(MOCK_CONTEXT);
+
             // assert model generation succeeds with a single schema file
             expect(graphqlCodegen.codegen).toBeCalled();
+
             // assert model files are generated in expected output directory
             expect(fs.readdirSync(outputDirectory).length).toBeGreaterThan(0);
         });
 
         it(frontend + ': Should generate models from any subdirectory in schema folder', async () => {
             // mock the input and output file structure
             const schemaFolderPath = path.join(MOCK_BACKEND_DIRECTORY, 'api', MOCK_PROJECT_NAME, 'schema', 'nested', 'deeply');
+            const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
             const mockedFiles = {};
             mockedFiles[schemaFolderPath] = {
                 'myschema.graphql': ' type SimpleModel { id: ID! status: String } '
             };
-            mockedFiles[path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend])] = {}
-            mock(mockedFiles);
+            mockedFiles[outputDirectory] = {}
+            mockFs(mockedFiles);
             MOCK_CONTEXT.amplify.getProjectConfig.mockReturnValue({frontend: frontend});
-            const outputDirectory = path.join(MOCK_PROJECT_ROOT, OUTPUT_PATHS[frontend]);
+
             // assert empty folder before generation
             expect(fs.readdirSync(outputDirectory).length).toEqual(0);
+
             await generateModels(MOCK_CONTEXT);
+
             // assert model generation succeeds with a single schema file
             expect(graphqlCodegen.codegen).toBeCalled();
+
             // assert model files are generated in expected output directory
             expect(fs.readdirSync(outputDirectory).length).toBeGreaterThan(0);
         });
     };
 
-    afterEach(mock.restore);
+    afterEach(mockFs.restore);
+});
+
+describe('codegen models respects cleanGeneratedModelsDirectory', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    MOCK_CONTEXT.amplify.getEnvInfo.mockReturnValue({projectPath: MOCK_PROJECT_ROOT});
+    MOCK_CONTEXT.amplify.getResourceStatus.mockReturnValue(
+          {
+            allResources: [
+                {
+                    service: 'AppSync', 
+                    providerPlugin: 'awscloudformation',
+                    resourceName: MOCK_PROJECT_NAME
+                  }
+              ]
+          }
+      );
+    MOCK_CONTEXT.amplify.executeProviderUtils.mockReturnValue([]);
+    MOCK_CONTEXT.amplify.pathManager.getBackendDirPath.mockReturnValue(MOCK_BACKEND_DIRECTORY);
+    graphqlCodegen.codegen.mockReturnValue(MOCK_GENERATED_CODE);
```

Comment:
The beforeEach in two test suites(line52-69 and line125-142) seems to have the same code/mock values. Can you move the duplicate mocking to the beginning?

Reply:
Refactored it to a method. Thanks